xmovie/core.py

```diff
@@ -358,12 +370,62 @@ def save_frames(self, odir, progress=False):
         elif ~tqdm_avail and progress:
             warnings.warn("Cant show progess bar at this point. Install tqdm")
 
-        for fi in frame_range:
-            fig, ax, pp = self.render_frame(fi)
-            frame_save(
-                fig, fi, odir=odir, frame_pattern=self.frame_pattern, dpi=self.dpi
+        for timestep in frame_range:
+            fig, ax, pp = self.render_single_frame(timestep)
+            save_single_frame(
+                fig, timestep, odir=odir, frame_pattern=self.frame_pattern, dpi=self.dpi
             )
 
+
+
+    def save_frames_parallel(self, odir):
+        '''
+        Saves all frames in parallel using dask.map_blocks.
+
+        Parameters
+        ----------
+        odir : path
+            path to output directory
+        '''
+        import numpy as np
+        import dask.array as darray
+
+        da = self.data
+        framedim = self.framedim
+        
+        assert isinstance(da, xr.DataArray)
+        
+        # move 'framedim' to be last dimension if not already
+        if da.dims[-1] != framedim:
+            new_dims = list(da.dims)
+            new_dims.append(new_dims.pop(new_dims.index(framedim))) # Move framedim to last axis
+            da = da.transpose(..., framedim)
+
+        da = da.chunk({self.framedim : 1}) # DataArray needs to have length-1 chunks in framedim
+        chunk_dims = [ dim for dim in da.dims if dim != framedim ] # Create dimensions for each chunk
+               
+
+        total_time = da[framedim]
+        def create_render_save_single_frame2(xr_array, framedim):
+            time_of_chunk = xr_array[framedim]
+            timestep = abs(total_time - time_of_chunk[0]).argmin().item() # get index of chunk in framedim
+
+            fig, ax, pp = self.render_single_frame(timestep)
+            save_single_frame(fig, timestep, odir=odir, frame_pattern=self.frame_pattern, dpi=self.dpi)
+
+            return time_of_chunk
+
+        mapped_save_and_close_frames = da.map_blocks(func=create_render_save_single_frame2,
+                                                     args=(framedim,),
+                                                     template=xr.ones_like(da[framedim]).chunk({framedim:1}),
+                                                     )
+        mapped_save_and_close_frames.compute(scheduler="single-threaded")
```

Comment:
Is this just for testing purposes? In my understanding this will reduce the calculation to a for loop, which could explain that there is no speedup. Apologies if I missed earlier discussions why this is needed.

Reply:
Yeah, sorry, I forgot to change this back when I push the commit! Like I said I wanna have a `compute_kwargs` in `Movie.save` that will go there. I'll implement that in a bit